icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -211,18 +268,34 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         int correctedNumBreaks = 0;
         for (int i = numBreaks - 1; i >= 0; i--) {
             int pos = charPositions[t_boundary[i]] + startPos;
-            if (!(foundBreaks.contains(pos) || pos == startPos)) {
+            if (!foundBreaks.contains(pos)
+                    && (pos != startPos
+                    || pos ==0
+                    || (pos == startPos  && isClosePunct(inText.setIndex(pos-1))))) {
                 foundBreaks.push(charPositions[t_boundary[i]] + startPos);
                 correctedNumBreaks++;
             }
         }
 
-        if (!foundBreaks.isEmpty() && foundBreaks.peek() == endPos) {
+        if (!foundBreaks.isEmpty()
```

Comment:
For word break, even if the logic here pops the ending breakpoint, it will be added back in 
https://github.com/unicode-org/icu/blob/main/icu4j/main/classes/core/src/com/ibm/icu/text/RuleBasedBreakIterator.java#L1231

To keep the same logic for word breaking, I add isPhraseBreaking but actually no impact for word break.